File: android/core/src/main/kotlin/os/blazen/jessica/core/IntentMatch.kt
package os.blazen.jessica.core

/**
 * One match result returned by [JessicaCore.matchIntent].
 *
 * Mirrors the `IntentMatch` shape in `domains/jessica-core/src/intent.rs`
 * so a JSON round-trip across JNI is lossless. Treat this file as part of
 * the Rust ↔ Kotlin contract — changes require the matching Rust update.
 */
data class IntentMatch(
    val name: String,
    val language: String,
    val action: String,
    val tool: String? = null,
    val params: Map<String, String> = emptyMap(),
    val confirm: Confirm = Confirm.NEVER,
) {
    enum class Confirm { NEVER, SOFT, HARD }
}

File: ios/JessicaCore/Sources/JessicaCore/IntentMatch.swift
import Foundation

/// One match result returned by ``JessicaCore/matchIntent(transcript:language:)``.
///
/// Mirrors the `IntentMatch` shape in
/// `domains/jessica-core/src/intent.rs` so a JSON round-trip
/// across the FFI is lossless. Treat this struct as part of the
/// Rust ↔ Swift contract — changes here require the matching Rust update.
public struct IntentMatch: Equatable, Sendable, Codable {
    public enum Confirm: String, Sendable, Codable {
        case never
        case soft
        case hard
    }

    public let name: String
    public let language: String
    public let action: String
    public let tool: String?
    public let params: [String: String]
    public let confirm: Confirm

    public init(
        name: String,
        language: String,
        action: String,
        tool: String? = nil,
        params: [String: String] = [:],
        confirm: Confirm = .never
    ) {
        self.name = name
        self.language = language
        self.action = action
        self.tool = tool
        self.params = params
        self.confirm = confirm
    }
}
